### PR TITLE
Add documentation examples for ``overridden-final-method`` checker

### PR DESCRIPTION
**Pull request by mbyrnepr2**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
|    | :bug: Bug fix          |
|    | :sparkles: New feature |
|    | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Partially closes issue https://github.com/PyCQA/pylint/issues/5953

### CODE REVIEW

**Review by DanielNoord — APPROVED** [review]

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by mbyrnepr2** [comment]
Thanks for the reviews @DanielNoord @Pierre-Sassoulas. I need to revisit since the CI has found an issue.

**Comment by DanielNoord** [comment]
Could be a check-messages issue again..

**Comment by mbyrnepr2** [comment]
> Could be a check-messages issue again..

Thanks. I had the same thought since reading the issue created by Andreas.

**Comment by Pierre-Sassoulas** [comment]
I did not expect https://github.com/PyCQA/pylint/pull/5956 to catch so many hard to find bug, but this PR has already been well worth the hassle.

**Comment by DanielNoord** [comment]
This is interesting... Are we getting a crash because we previously never entered that function?

**Comment by mbyrnepr2** [comment]
> 

The visit function? I imagine that has been working ok since reproducing the emitted warning is as expected in the functional tests. Or is this something else? I haven't dug into what the current failing test in the CI is checking, does it expect every possible message to be present in the decorator?

**Comment by mbyrnepr2** [comment]
@DanielNoord I don't see anything obviously incorrect with respect to the setup of the various `check_messages` decorators in the code.

I've noticed that locally on my Windows machine the test which is failing in the CI passes. Instead I see this one:
```python
    def _runTest(self) -> None:
        """Run the test and assert message differences."""
        self._linter.check([str(self._test_file[1])])
        expected_messages = self._get_expected()
        actual_messages = self._get_actual()
>       assert expected_messages == actual_messages
E       AssertionError: assert Counter({(1, ...builtin'): 1}) == Counter()
E         Left contains 1 more item:
E         {(1, 'bad-builtin'): 1}
E         Use -v to get more diff

pylint\doc\test_messages_documentation.py:111: AssertionError
=========================== short test summary info ===========================
FAILED pylint\doc\test_messages_documentation.py::test_code_examples[bad-builtin-bad]
======================== 1 failed, 97 passed in 20.53s ========================

```

Failing CI test:
```python
def _runTest(self) -> None:
        """Run the test and assert message differences."""
        self._linter.check([str(self._test_file[1])])
        expected_messages = self._get_expected()
        actual_messages = self._get_actual()
>       assert expected_messages == actual_messages
E       AssertionError: assert Counter({(11,...-method'): 1}) == Counter()
E         Left contains 1 more item:
E         {(11, 'overridden-final-method'): 1}
E         Full diff:
E         - Counter()
E         + Counter({(11, 'overridden-final-method'): 1})

doc/test_messages_documentation.py:111: AssertionError
=========================== short test summary info ============================
FAILED doc/test_messages_documentation.py::test_code_examples[overridden-final-method-bad]
======================== 1 failed, 97 passed in 10.[44](https://github.com/PyCQA/pylint/runs/5799254136?check_suite_focus=true#step:6:44)s =========================
```

**Comment by DanielNoord** [comment]
@mbyrnepr2
Could this be the issue:
```console
❯ pylint /Users/daniel/DocumentenLaptop/Programming/Github/pylint/doc/data/messages/o/overridden-final-method/bad.py
************* Module bad
doc/data/messages/o/overridden-final-method/bad.py:5:5: W2602: typing.final is not supported by all versions included in the py-version setting (using-final-decorator-in-unsupported-version)

------------------------------------------------------------------
Your code has been rated at 8.57/10 (previous run: 0.00/10, +8.57)
```

**Comment by DanielNoord** [comment]
@mbyrnepr2
Could this be the issue:
```console
❯ pylint /Users/daniel/DocumentenLaptop/Programming/Github/pylint/doc/data/messages/o/overridden-final-method/bad.py
************* Module bad
doc/data/messages/o/overridden-final-method/bad.py:5:5: W2602: typing.final is not supported by all versions included in the py-version setting (using-final-decorator-in-unsupported-version)

------------------------------------------------------------------
Your code has been rated at 8.57/10 (previous run: 0.00/10, +8.57)
```

File: doc/data/messages/o/overridden-final-method/bad.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+class Base:
+    @final
+    def my_method(self):
+        pass
+
+
+class Subclass(Base):
+    def my_method(self):  # [overridden-final-method]
+        pass
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/o/overridden-final-method/bad.py:12]:
```suggestion
class Animal:
    @final
    def breath(self):
        return True


class Cat(Animal):
    def breath(self):  # [overridden-final-method]
        pass
```

File: doc/data/messages/o/overridden-final-method/good.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+class Base:
+    @final
+    def my_method(self):
+        pass
+
+
+class Subclass(Base):
+    def my_other_method(self):
+        pass
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/o/overridden-final-method/good.py:12]:
```suggestion
class Animal:
    @final
    def breath(self):
        return True


class Cat(Animal):
    def purr(self):
        return True
```

**Review by Pierre-Sassoulas — COMMENTED** [review]
Animal example. It was either that or a fruits example 😄

File: doc/data/messages/o/overridden-final-method/bad.py (hunk 1 above)
Comment (mbyrnepr2) [reply doc/data/messages/o/overridden-final-method/bad.py]: Quite an intuitive example @Pierre-Sassoulas.
I've changed the naming slightly.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**